packages/workflow/src/internals.ts

```diff
@@ -651,8 +677,10 @@ export class Activator implements ActivationHandler {
         this.rejectUpdate(protocolInstanceId, error);
         return;
       }
-      const execute = composeInterceptors(this.interceptors.inbound, 'handleUpdate', this.updateNextHandler.bind(this));
       this.acceptUpdate(protocolInstanceId);
+      const execute = composeInterceptors(this.interceptors.inbound, 'handleUpdate', this.updateNextHandler.bind(this));
+      const { unfinishedPolicy } = entry;
+      this.inProgressUpdates.set(updateId, { name, unfinishedPolicy, id: updateId });
       const res = execute(input)
         .then((result) => this.completeUpdate(protocolInstanceId, result))
```

Comment:
Combine the `.then` and `.catch` clauses in a single call to `.then`. That will create a single promise instead of two. Then make sure to `untrackPromise()` that promise, before calling `.finally()` on it (which also needs to be `untrackPromise()`, but that one is already there).

Reply:
Let's address this in https://github.com/temporalio/sdk-typescript/issues/1486